build_system/src/test.rs

```diff
@@ -1218,6 +1247,7 @@ fn run_all(env: &Env, args: &TestArg) -> Result<(), String> {
     test_libcore(env, args)?;
     extended_sysroot_tests(env, args)?;
     test_rustc(env, args)?;
+    cargo_tests(env, args)?;
```

Comment:
Is this ran? I'm asking because `test_rustc` might return a non-zero exit code, so perhaps the `?` will return from this function.